### PR TITLE
fix insert right SQL types values (native types), non stringify 

### PR DESCRIPTION
**Pull request by AlvaroVega**
issue https://github.com/telefonicaid/fiware-cygnus/issues/1661
previous PR https://github.com/telefonicaid/fiware-cygnus/pull/1663
- [x] insert NULL instead empty string
- [x] insert Number without stingify

As in ppstgresql and postgis sink, attr_native_types option is added

### CODE REVIEW

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java [hunk 1]
```diff
@@ -384,11 +384,17 @@ public String getValuesForInsert() {
...
+                    } else {
+                        value = "'" + value + "'";
+                    }
                     if (first) {
-                        valuesForInsert += "'" + values.get(i) + "'";
+                        valuesForInsert += value;
                         first = false;
                     } else {
-                        valuesForInsert += ",'" + values.get(i) + "'";
+                        valuesForInsert += "," + value;;
```
Comment (mrutid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java]: extra semicolon?

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java [hunk 2]
```diff
@@ -384,11 +384,17 @@ public String getValuesForInsert() {
             
                 while (it.hasNext()) {
                     ArrayList<String> values = (ArrayList<String>) aggregation.get((String) it.next());
+                    String value = values.get(i);
+                    if (!value || value.equals("")) {
+                        value = "NULL";
+                    } else {
+                        value = "'" + value + "'";
```
Comment (mrutid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java]: what if value is a number?

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java (hunk 1 above)
Comment (AlvaroVega) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java]: fixed in 007a052b59e3d7e08ebb3932fa1e26f82e1b2e51

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java (hunk 2 above)
Comment (AlvaroVega) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java]: That check was done for Postgresql and Postgis, but not for MySQL:
https://github.com/telefonicaid/fiware-cygnus/pull/1663/files

Comment (AlvaroVega) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java]: fixed in 938a871a3c56b1353e1b99801fd70aac4c1e630a

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java [hunk 3]
```diff
@@ -384,11 +406,18 @@ public String getValuesForInsert() {
             
                 while (it.hasNext()) {
                     ArrayList<String> values = (ArrayList<String>) aggregation.get((String) it.next());
+                    String valueType = (String) it.next();
+                    String value = values.get(i);
+                    if (value == null) { // TBD check also if value == "" ?
+                        value = "NULL";
+                    } else if (!(attrNativeTypes && valueType.equals("Number"))) {
```
Comment (mrutid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java]: Its not just a matter of NGSI Type, but also related to REAL Type (NGSIv2 JSON native types).

**Pull request merged**